conda_build_all/tests/unit/test_version_matrix.py

```diff
@@ -106,6 +106,83 @@ def test_dependency_on_py27(self):
     #                            ])
     #                     )
 
+    def test_numpy_xx(self):
+        """
+        Check whether the numpy x.x version spec generates the right cases.
+        """
+
+        # Build an index that contains numpy 1.9 and 1.10 on python 2.7 and
+        # 3.5 for a total of 4 numpy/python combinations.
+        pythons = ['2.7', '3.5']
+        # Only major/minor in the numpy list here because that is what is used
+        # for the build matrix.
+        numpys = ['1.9', '1.10']
+        for python_version in pythons:
+            python_build_string = 'py' + python_version.replace('.', '')
+            self.index.add_pkg('python', python_version)
+            for numpy_version in numpys:
+                # Add a patch version to each numpy since that is how the
+                # versions are numbered.
+                self.index.add_pkg('numpy',
+                                   numpy_version + '.2',
+                                   python_build_string,
+                                   depends=['python ' + python_version])
+
+        # Cases to check
+        numpy_dep_cases = []
+        expect_result = []
+
+        # Case 1:
+        #   Only a numpy x.x spec, expect all four cases to be in the matrix.
+
+        # Add case.
+        numpy_dep_cases.append(('numpy x.x', 'python'))
+        # Generate expected results for this case.
+        results = []
+        for python in pythons:
+            for numpy in numpys:
+                results.append((('python', python), ('numpy', numpy)))
+
+        expect_result.append(results)
```

Comment:
Personally I would have made these separate test methods so that it is easy to isolate individual tests for debugging, but they are covering the right things, so :+1: